### ISSUE TITLE
Unable to export container

### ISSUE THREAD

**Issue by maltyyev** [docker/for-mac#2679]
### Expected behavior
Export container to mysql.tar
### Actual behavior

> Error response from daemon: Bad response from Docker engine

### Information

Docker for Mac: version: 17.12.0-ce-mac55 (18467c0ae7afb7a736e304f991ccc1a61d67a4ab)
macOS: version 10.12.6 (build: 16G1212)
logs: /tmp/5EC486D3-0587-483C-A3E7-8024862CBD91/20180309-011820.tar.gz
[OK]     vpnkit
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     app
[OK]     virtualization VT-X
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     moby-console
[OK]     osxfs
[OK]     logs
[OK]     docker-cli
[OK]     disk
Failure: Could not upload diagnostic data to remote server (docker-diagnose exit code is 1)

You can also run diagnostics manually :
$ /Applications/Docker.app/Contents/Resources/bin/docker-diagnose -u

### Steps to reproduce the behavior

  1. In Terminal enter:
`docker container export mysql > mysql.tar`
  2. See the error response

I must say I didn't have this issue about 2 days ago, before I updated Docker.

**Comment by guillaumerose** [comment]
I can confirm this issue. Here is the dockerd stack trace:

```
2018-03-12 12:37:17.409021 I | http: panic serving @: runtime error: invalid memory address or nil pointer dereference
goroutine 7405 [running]:
net/http.(*conn).serve.func1(0xc420c099a0)
	/usr/local/go/src/net/http/server.go:1697 +0xd2
panic(0x21655c0, 0x2e71f70)
	/usr/local/go/src/runtime/panic.go:491 +0x287
github.com/docker/docker/daemon.(*Daemon).containerExport(0xc420448240, 0xc420448480, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/export.go:69 +0x31a
github.com/docker/docker/daemon.(*Daemon).ContainerExport(0xc420448240, 0xc4202480d6, 0xc, 0x2e9e020, 0xc42137e000, 0x0, 0x0)
	/go/src/github.com/docker/docker/daemon/export.go:37 +0x2f4
github.com/docker/docker/api/server/router/container.(*containerRouter).getContainersExport(0xc4201343c0, 0x7fcfcd3a53c8, 0xc421b62e40, 0x2eb8be0, 0xc42137e000, 0xc42140e500, 0xc421b62d50, 0x190b3c6, 0x4)
	/go/src/github.com/docker/docker/api/server/router/container/container_routes.go:159 +0xd3
github.com/docker/docker/api/server/router/container.(*containerRouter).(github.com/docker/docker/api/server/router/container.getContainersExport)-fm(0x7fcfcd3a53c8, 0xc421b62e40, 0x2eb8be0, 0xc42137e000, 0xc42140e500, 0xc421b62d50, 0x7fcfcd3a53c8, 0xc421b62e40)
	/go/src/github.com/docker/docker/api/server/router/container/container.go:37 +0x6b
github.com/docker/docker/api/server/middleware.ExperimentalMiddleware.WrapHandler.func1(0x7fcfcd3a53c8, 0xc421b62e40, 0x2eb8be0, 0xc42137e000, 0xc42140e500, 0xc421b62d50, 0x7fcfcd3a53c8, 0xc421b62e40)
	/go/src/github.com/docker/docker/api/server/middleware/experimental.go:27 +0xda
github.com/docker/docker/api/server/middleware.VersionMiddleware.WrapHandler.func1(0x7fcfcd3a53c8, 0xc421b62db0, 0x2eb8be0, 0xc42137e000, 0xc42140e500, 0xc421b62d50, 0xc420248b00, 0xc422326000)
	/go/src/github.com/docker/docker/api/server/middleware/version.go:62 +0x425
github.com/docker/docker/pkg/authorization.(*Middleware).WrapHandler.func1(0x7fcfcd3a53c8, 0xc421b62db0, 0x2eb8be0, 0xc42137e000, 0xc42140e500, 0xc421b62d50, 0x10, 0x401557)
	/go/src/github.com/docker/docker/pkg/authorization/middleware.go:59 +0x83d
github.com/docker/docker/api/server/middleware.DebugRequestMiddleware.func1(0x7fcfcd3a53c8, 0xc421b62db0, 0x2eb8be0, 0xc42137e000, 0xc42140e500, 0xc421b62d50, 0x7fcfcd3a53c8, 0xc421b62db0)
	/go/src/github.com/docker/docker/api/server/middleware/debug.go:22 +0x180
github.com/docker/docker/api/server.(*Server).makeHTTPHandler.func1(0x2eb8be0, 0xc42137e000, 0xc42140e500)
	/go/src/github.com/docker/docker/api/server/server.go:137 +0x1c8
net/http.HandlerFunc.ServeHTTP(0xc420bb9260, 0x2eb8be0, 0xc42137e000, 0xc42140e500)
	/usr/local/go/src/net/http/server.go:1918 +0x46
github.com/docker/docker/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420b27e00, 0x2eb8be0, 0xc42137e000, 0xc42140e500)
	/go/src/github.com/docker/docker/vendor/github.com/gorilla/mux/mux.go:103 +0x22d
github.com/docker/docker/api/server.(*routerSwapper).ServeHTTP(0xc420db5f00, 0x2eb8be0, 0xc42137e000, 0xc42140e500)
	/go/src/github.com/docker/docker/api/server/router_swapper.go:29 +0x72
net/http.serverHandler.ServeHTTP(0xc420335790, 0x2eb8be0, 0xc42137e000, 0xc42140e500)
	/usr/local/go/src/net/http/server.go:2619 +0xb6
net/http.(*conn).serve(0xc420c099a0, 0x2eba8a0, 0xc421258b00)
	/usr/local/go/src/net/http/server.go:1801 +0x71f
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2720 +0x28a
```

**Comment by thaJeztah** [comment]
@guillaumerose what version were you able to reproduce this on? Docker 18.03.0-rc1 on Mac seems to work for me; any particular container/image?

**Comment by thaJeztah** [comment]
Based on the panic (heard @guillaumerose was on 18.03.0-ce-rc3), I suspect `container.BaseFS` is nil somehow; https://github.com/docker/docker-ce/blob/v18.03.0-ce-rc3/components/engine/daemon/export.go#L69

**Comment by thaJeztah** [comment]
I see in `daemon.go` there's a check for this; https://github.com/moby/moby/blame/1790ce52e96fd66621b5d8eb2e203203874a7921/daemon/daemon.go#L1056

**Comment by guillaumerose** [comment]
After a reset factory, it works. Is it due to a version upgrade ?

**Comment by guillaumerose** [comment]
@maltyyev Can you retry to make a diagnostic ? The upload failed for the one you mention.

**Comment by maltyyev** [comment]
@guillaumerose I'm sorry, it doesn't upload, I don't know why. Is there any other way I can send you diagnostic report?

**Comment by vdemeester** [comment]
Upstream issue for that : https://github.com/moby/moby/issues/36561 :angel:

**Comment by thaJeztah** [comment]
The only location this value is set is in [daemon/daemon.go#L1066](https://github.com/moby/moby/blob/1790ce52e96fd66621b5d8eb2e203203874a7921/daemon/daemon.go#L1066), and that part of the code won't be reached if the container's RWLayer was not set, or failed to mount [daemon/daemon.go#L1047-L1053](https://github.com/moby/moby/blob/1790ce52e96fd66621b5d8eb2e203203874a7921/daemon/daemon.go#L1047-L1053)

I do see one code path where `daemon.Mount()` can fail, and the error is ignored (only a warning is printed); [daemon/daemon.go#L294-L301](https://github.com/moby/moby/blob/1790ce52e96fd66621b5d8eb2e203203874a7921/daemon/daemon.go#L294-L301)

**Comment by maltyyev** [comment]
Ok, reset to factory worked for me too. But now I can't get an image, got unexpected EOF instead =\ This is a really strange situation.
Here is the Terminal output:
`docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
ca4f61b1923c: Pulling fs layer 
docker: error pulling image configuration: Get https://dseasb33srnrn.cloudfront.net/registry-v2/docker/registry/v2/blobs/sha256/f2/f2a91732366c0332ccd7afd2a5c4ff2b9af81f549370f7a19acd460f87686bc7/data?Expires=1520909211&Signature=GP26HrVDhBARRYod1DFUtqo8HfqhziXBEFVvczr-tzboWHk-dIe4tVdtSvDZn1329qvTEdL5SYYAy6lB4LL1NO7v7BkX-sSRNYpuPDbWpkNx0PDFPI66gqhNXJSEYHe7s6~kG3RuYiCl2pJhq80iNkB9XELTODACy9F20WBAGe8_&Key-Pair-Id=APKAJECH5M7VWIS5YZ6Q: EOF.
See 'docker run --help'.`

**Comment by jimmont** [comment]
also cannot export from latest stable:
```
> docker container export mssql0 > mssql0-1.tar
Error response from daemon: Bad response from Docker engine
> docker -v
Docker version 17.12.0-ce, build c97c6d6
> system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 10.13.3 (17D102)
      Kernel Version: Darwin 17.4.0
```

**Comment by vdemeester** [comment]
https://github.com/moby/moby/issues/36561#issuecomment-373305522 — fix should be on 18.03 GA :angel:
